diff --git a/suckless/suckless/dwm/config.def.h b/suckless/suckless/dwm/config.def.h
index 24f153b..7262b60 100644
--- a/suckless/suckless/dwm/config.def.h
+++ b/suckless/suckless/dwm/config.def.h
@@ -158,8 +158,8 @@ static Key keys[] = {
   { MODKEY|ShiftMask,                       XK_k,                        focusstackhid,         {.i = -1 } },
   { MODKEY|ShiftMask,                       XK_minus,                    incnmaster,            {.i = +1 } },
   { MODKEY,                                 XK_d,                        incnmaster,            {.i = -1 } },
-  { MODKEY,                                 XK_h,                        setmfact,              {.f = -0.05} },
-  { MODKEY,                                 XK_l,                        setmfact,              {.f = +0.05} },
+  { MODKEY,                                 XK_h,                        setmfact,              {.f = +0.05} },
+  { MODKEY,                                 XK_l,                        setmfact,              {.f = -0.05} },
   { MODKEY,                                 XK_Tab,                      view,                  {0} },
   { MODKEY|ShiftMask,                       XK_c,                        killclient,            {0} },
   { MODKEY,                                 XK_c,                        killclient,            {0} },
diff --git a/suckless/suckless/dwm/dwm.c b/suckless/suckless/dwm/dwm.c
index bff37f3..78ae610 100644
--- a/suckless/suckless/dwm/dwm.c
+++ b/suckless/suckless/dwm/dwm.c
@@ -2263,12 +2263,12 @@ tile(Monitor *m)
 	for (i = 0, my = ty = m->gappx, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
 		if (i < m->nmaster) {
 			h = (m->wh - my) / (MIN(n, m->nmaster) - i) - m->gappx;
-			resize(c, m->wx + m->gappx, m->wy + my, mw - (2*c->bw) - m->gappx, h - (2*c->bw), 0);
+			resize(c, m->wx + m->gappx + m->ww - mw, m->wy + my, mw - (2*c->bw) - m->gappx, h - (2*c->bw), 0);
 			if (my + HEIGHT(c) < m->wh)
 				my += HEIGHT(c) + m->gappx;
 		} else {
 			 h = (m->wh - ty) / (n - i) - m->gappx;
-			 resize(c, m->wx + mw + m->gappx, m->wy + ty, m->ww - mw - (2*c->bw) - 2*m->gappx, h - (2*c->bw), 0);
+			 resize(c, m->wx + m->gappx, m->wy + ty, m->ww - mw - (2*c->bw) - 2*m->gappx, h - (2*c->bw), 0);
 			if (ty + HEIGHT(c) < m->wh)
 				ty += HEIGHT(c) + m->gappx;
 		}
